packages/netlify-cms-core/src/constants/configSchema.js

```diff
@@ -3,6 +3,36 @@ import { select, uniqueItemProperties, instanceof as instanceOf } from 'ajv-keyw
 import ajvErrors from 'ajv-errors';
 import { formatExtensions, frontmatterFormats, extensionFormatters } from 'Formats/formats';
 import { getWidgets } from 'Lib/registry';
+import { I18N_STRUCTURE, I18N_FIELD } from '../lib/i18n';
+
+const i18n = {
+  type: 'object',
+  properties: {
+    structure: { type: 'string', enum: Object.values(I18N_STRUCTURE) },
+    locales: {
+      type: 'array',
+      minItems: 2,
+      items: { type: 'string' },
```

Comment:
Locale strings containing unwanted characters like `.`  could break the multiple files mode here https://github.com/netlify/netlify-cms/pull/4139/files#diff-57dc49557fe997703a84eff0dc845cf3R100 that is one of the reasons for the hardcoded locales list. Maybe sanitize.

Reply:
Added validation here https://github.com/netlify/netlify-cms/pull/4139/commits/1fb2970876fe7b344bb5945e7e69ae533eddd33f